### PR TITLE
Ae 800/include assets in report

### PR DESCRIPTION
**Pull request by jonasfuegen**
Added assets to component vulnerabilites section and created a section with assets listing contained components. Added links to ease traversal.

### CODE REVIEW

File: plugins/ae-inventory-maven-plugin/src/main/java/org/metaeffekt/core/maven/inventory/mojo/PomReportCreationMojo.java [hunk 1]
```diff
@@ -131,7 +131,7 @@ private boolean skipExecution() {
         return false;
     }
 
-    private Inventory createInventoryFromPom() {
+    private Inventory  createInventoryFromPom() {
```
Comment (YanWittmann) [inline plugins/ae-inventory-maven-plugin/src/main/java/org/metaeffekt/core/maven/inventory/mojo/PomReportCreationMojo.java]: I love this additional space that was added here

File: libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/report/adapter/AssetReportAdapter.java [hunk 2]
```diff
@@ -41,6 +43,11 @@ public List<AssetMetaData> listAssets() {
         return assetMetaData;
     }
 
+    public List<Artifact> getRelatedArtifacts(AssetMetaData assetMetaData) {
+        return new ArrayList<>(InventoryUtils.getArtifactsForAsset(inventory, assetMetaData));
+
```
Comment (YanWittmann) [inline libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/report/adapter/AssetReportAdapter.java]: Why such empty line

File: libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/report/model/aeaa/AeaaVulnerabilityContextInventory.java [hunk 3]
```diff
@@ -431,6 +432,8 @@ protected void parse(List<Artifact> artifacts, List<VulnerabilityMetaData> vmds,
                         final List<Artifact> artifactsForVulnerability = artifactEntry.getValue();
                         for (Artifact artifact : artifactsForVulnerability) {
                             vulnerability.manuallyAffectsArtifact(key, artifact);
+                            vulnerability.manuallyAffectsAssets(InventoryUtils.getAssetsForArtifact(inventory,
```
Comment (YanWittmann) [inline libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/report/model/aeaa/AeaaVulnerabilityContextInventory.java]: - If you change something in a class prefixed with `Aeaa`, you will have to find the corresponding file in the [Artifact Analysis project](https://github.com/org-metaeffekt/metaeffekt-artifact-analysis/blob/429b60a1103db79ad083db179561d7d19e689107/modules/ae-artifact-analysis/src/main/java/com/metaeffekt/mirror/contents/base/VulnerabilityContextInventory.java#L44) to update as well.
- The same goes for the `Vulnerability` and it's superclasses.

File: libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/report/model/aeaa/AeaaMatchableDetailsAmbDataClass.java [hunk 4]
```diff
@@ -30,529 +27,603 @@
...
+	 * rather
+	 * as references from the artifact to this entry.
+	 * <pre>
+	 * Key:   Artifact field name
+	 * Value: Set&lt;Artifact&gt; affected artifacts
+	 * </pre>
+	 */
+	protected final Map<String, Set<Artifact>> affectedArtifacts = new HashMap<>();
+
+	protected final Set<AssetMetaData> affectedAssets = new HashSet<>();
```
Comment (YanWittmann) [inline libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/report/model/aeaa/AeaaMatchableDetailsAmbDataClass.java]: This information should not be stored in a property, but rather recalculated upon any access to the vulnerability instance using the `affectedArtifacts`.
Storing this information would surely lead to sync-issues in the future, having to always update the information in two properties (redundancy leads to inconsistency).

Also, if we were to update this file, the according class in the artifact analysis project would have to be updated as well.

File: libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/report/model/aeaa/AeaaMatchableDetailsAmbDataClass.java
Comment (YanWittmann) [inline libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/report/model/aeaa/AeaaMatchableDetailsAmbDataClass.java:1]: Please do not generally apply a new auto-formatter to an entire existing file.
As this file is a mere copy of a file in artifact analysis, it is important that the same formatting remains, as it allows for easily comparing the two for changes made in either of them to apply them to the other.

Comment (YanWittmann) [reply libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/report/model/aeaa/AeaaMatchableDetailsAmbDataClass.java:1]: What I want to say is, please restore the original formatting and only change the lines you need to.

File: libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/report/adapter/VulnerabilityReportAdapter.java
Comment (YanWittmann) [inline libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/report/adapter/VulnerabilityReportAdapter.java:1]: See my comment on the class `AeaaMatchableDetailsAmbDataClass` regarding the formatting.

File: libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/InventoryUtils.java
Comment (YanWittmann) [inline libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/InventoryUtils.java:1]: Please see my comment on `AeaaMatchableDetailsAmbDataClass`.

File: libraries/ae-inventory-processor/src/test/java/org/metaeffekt/core/inventory/processor/report/AssessmentInventoryMergerTest.java
Comment (YanWittmann) [inline libraries/ae-inventory-processor/src/test/java/org/metaeffekt/core/inventory/processor/report/AssessmentInventoryMergerTest.java:1]: Please see `AssessmentInventoryMergerTest`. Also, we do not use tabs for indentation, but rather spaces.

**Review by YanWittmann — CHANGES_REQUESTED** [review]
Okay, I think we're going to chat a little about these changes to align them with the rest of the codebase and a couple of other things.

**Comment by karsten-klein** [comment]
Decided to cherry-pick and review. Relevant changes have been merged to core

**Pull request closed without merging**